packages/vue/src/auth/mapAuthState.ts

```diff
@@ -53,44 +61,132 @@ export const connectMapState = (_this: any) => {
     mapTeamState: mapTeamState(_this),
     mapTenantsState: mapTenantsState(_this),
   });
+};
+
+const actionGetter = (action: any) => (...args: any) => (ActionsHolder.getAction(action) as any)(...args);
 
+type MapLoginActions = LoginActions & { loginWithRedirect: LoginActions['requestHostedLoginAuthorize'] };
+export const mapAuthActions = <K extends keyof AuthActions>(action: K): AuthActions[K] => actionGetter(action);
+export const mapLoginActions = <K extends keyof MapLoginActions>(action: K): MapLoginActions[K] => {
+  if (action === 'loginWithRedirect') {
+    return actionGetter('requestHostedLoginAuthorize');
+  }
+  return actionGetter(action);
 };
+export const mapAcceptInvitationActions = <K extends keyof AcceptInvitationActions>(
+  action: K
+): AcceptInvitationActions[K] => actionGetter(action);
+export const mapActivateAccountActions = <K extends keyof ActivateAccountActions>(
+  action: K
+): ActivateAccountActions[K] => actionGetter(action);
+export const mapApiTokensActions = <K extends keyof ApiTokensActions>(action: K): ApiTokensActions[K] =>
+  actionGetter(action);
+export const mapForgotPasswordActions = <K extends keyof ForgotPasswordActions>(action: K): ForgotPasswordActions[K] =>
+  actionGetter(action);
+export const mapMfaActions = <K extends keyof MfaActions>(action: K): MfaActions[K] => actionGetter(action);
+export const mapProfileActions = <K extends keyof ProfileActions>(action: K): ProfileActions[K] => actionGetter(action);
+export const mapSecurityPolicyActions = <K extends keyof SecurityPolicyActions>(action: K): SecurityPolicyActions[K] =>
+  actionGetter(action);
+export const mapSignupActions = <K extends keyof SignUpActions>(action: K): SignUpActions[K] => actionGetter(action);
+export const mapSocialLoginActions = <K extends keyof SocialLoginActions>(action: K): SocialLoginActions[K] =>
+  actionGetter(action);
+export const mapSsoActions = <K extends keyof SSOActions>(action: K): SSOActions[K] => actionGetter(action);
+export const mapTeamActions = <K extends keyof TeamActions>(action: K): TeamActions[K] => actionGetter(action);
+export const mapTenantsActions = <K extends keyof TenantsActions>(action: K): TenantsActions[K] => actionGetter(action);
 
+export const connectFronteggStoreV3 = (store: EnhancedStore) => {
+  const initialState = store.getState();
 
-const actionGetter = (action: any) => (...args: any) => (ActionsHolder.getAction(action) as any)(...args);
+  const authState = reactive({ ...initialState.auth });
+
+  const unsubscribe = store.subscribe(() => {
+    const state = store.getState().auth;
+
+    Object.entries(state).forEach(([key, value]) => {
+      authState[key as keyof AuthState] = value;
+    });
+  });
+
+  return { authState, unsubscribe };
+};
+
+export const useFronteggLoaded = () => {
+  const fronteggLoaded = inject('fronteggLoaded') as boolean;
```

Comment:
I would generally suggest using InjectionKeys instead of strings. It can also provide automatic typing so force casting with 'as' won't be needed
https://vuejs.org/guide/components/provide-inject.html#working-with-symbol-keys

Reply:
will go with InjectionKeys - good point